shared/search/providers.test.ts

```diff
@@ -437,6 +439,23 @@ describe('search providers', () => {
         })
     })
 
+    /** Create providers with the definition provider fed into itself. */
+    const recurProviders = (api: API): Providers => {
```

Comment:
Cute. This is roughly how parser combinators deal with recursive productions too. 🙂